autotest/test_geospatial_index.py

```diff
@@ -0,0 +1,663 @@
+# Test Tie-Breaking (Lowest Cell ID)
+# ============================================================================
+
+
+@pytest.mark.parametrize(
+    "grid_fixture",
+    [
+        "minimal_vertex_grid",
+        "minimal_unstructured_grid",
+    ],
+)
+def test_tiebreaker_lowest_id(grid_fixture, request):
+    """Test that boundary points return lowest cell ID."""
+    grid = request.getfixturevalue(grid_fixture)
+    index = GeospatialIndex(grid)
+
+    # Point on shared boundary between cells 0 and 1 (x=1.0)
+    assert index.query_point(1.0, 0.5) == 0
+
+
+# ============================================================================
+# Test 3D Layered UnstructuredGrid
+# ============================================================================
+
+
+def test_3d_query(layered_unstructured_grid):
+    """Test 3D point queries on layered unstructured grid."""
+    index = GeospatialIndex(layered_unstructured_grid)
+
+    # Test all 6 cells across 3 layers
+    test_cases = [
+        (0.5, 0.5, 8.5, 0),  # layer 0, left
+        (1.5, 0.5, 8.5, 1),  # layer 0, right
+        (0.5, 0.5, 5.5, 2),  # layer 1, left
+        (1.5, 0.5, 5.5, 3),  # layer 1, right
+        (0.5, 0.5, 2.5, 4),  # layer 2, left
+        (1.5, 0.5, 2.5, 5),  # layer 2, right
+    ]
+
+    for x, y, z, expected in test_cases:
+        assert index.query_point(x, y, z=z) == expected
+
+
+def test_3d_layer_boundary_tiebreaker(layered_unstructured_grid):
+    """Test tie-breaking at layer boundaries returns lowest cell ID."""
+    index = GeospatialIndex(layered_unstructured_grid)
+
+    # z=7.0: boundary between layer 0 (cell 0) and layer 1 (cell 2)
+    assert index.query_point(0.5, 0.5, z=7.0) == 0
+
+    # z=4.0: boundary between layer 1 (cell 2) and layer 2 (cell 4)
+    assert index.query_point(0.5, 0.5, z=4.0) == 2
+
+
+def test_3d_xy_boundary_tiebreaker(layered_unstructured_grid):
+    """Test tie-breaking at x/y boundaries in 3D grid."""
+    index = GeospatialIndex(layered_unstructured_grid)
+
+    # x=1.0 boundary in each layer
+    assert index.query_point(1.0, 0.5, z=8.5) == 0  # layer 0
+    assert index.query_point(1.0, 0.5, z=5.5) == 2  # layer 1
+    assert index.query_point(1.0, 0.5, z=2.5) == 4  # layer 2
+
+
+def test_3d_1to1_mapping(layered_unstructured_grid):
+    """Test 1:1 mapping for 3D vectorized queries."""
+    index = GeospatialIndex(layered_unstructured_grid)
+
+    x = np.array([0.5, 1.5, 0.5, 5.0])
+    y = np.array([0.5, 0.5, 0.5, 5.0])
+    z = np.array([8.5, 5.5, 2.5, 8.5])
+
+    cellids = index.query_points(x, y, z=z)
+
+    assert len(cellids) == 4
+    assert cellids[0] == 0
+    assert cellids[1] == 3
+    assert cellids[2] == 4
+    assert np.isnan(cellids[3])
+
+
+# ============================================================================
+# Test StructuredGrid Native Methods
+# ============================================================================
+
+
+def test_structured_boundary_tiebreaker(simple_structured_grid):
```

Comment:
if this doesn't use `GeospatialIndex` under the hood, it probably belongs in `test_grid.py`?